api.go

```diff
@@ -207,7 +219,18 @@ func (r *request) toHTTP() (*http.Request, error) {
 	}
 
 	// Create the HTTP request
-	return http.NewRequest(r.method, urlRaw, r.body)
+	req, err := http.NewRequest(r.method, urlRaw, r.body)
+
+	// Setup auth
+	if err == nil && r.config.HttpAuth != nil {
+		r.config.HttpAuth.auth(req)
+	}
+
+	return req, err
+}
+
+func (a *HttpBasicAuth) auth(req *http.Request) {
```

Comment:
Yeah in this case, it is simple enough that I prefer to keep the code clean. If we need abstractions, we can add them later.